test/unit/range/iterator_test_template.hpp

```diff
@@ -0,0 +1,502 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2019, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2019, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+// -----------------------------------------------------------------------------------------------------
+
+#pragma once
+
+#include <gtest/gtest.h>
+
+#include <seqan3/core/type_traits/range.hpp>
+#include <seqan3/std/iterator>
+#include <seqan3/std/ranges>
```

Comment:
I would prefer it if we can get rid of it. we simply could require the test range to have a begin and end member function.